base/common/src/main/java/org/dogtagpki/nss/NSSDatabase.java

```diff
@@ -1132,8 +1176,12 @@ public PKCS10 createPKCS10Request(
         logger.debug("NSSDatabase: - subject: " + subject);
 
         PK11PrivKey privateKey = (PK11PrivKey) keyPair.getPrivate();
-        String algorithm = hash + "with" + privateKey.getType();
-
+        String algorithm;
+        if (privateKey.getKeyType() == KeyType.MLDSA) {
```

Comment:
maybe add a little comment to help the future readers:
 // ignore hash for ML-DSA
// Returns "ML-DSA-44", "ML-DSA-65", etc.